pkg/costmodel/metrics.go

```diff
@@ -607,7 +613,11 @@ func (cmme *CostModelMetricsEmitter) Start() bool {
 			for labelString, seen := range loadBalancerSeen {
 				if !seen {
 					labels := getLabelStringsFromKey(labelString)
-					cmme.LBCostRecorder.DeleteLabelValues(labels...)
+					ok := cmme.LBCostRecorder.DeleteLabelValues(labels...)
```

Comment:
Without checking that deletion actually occurs, we just silently fail here and keep on truckin'.

Reply:
![bitmoji](https://sdk.bitmoji.com/render/panel/a7e6ab20-9788-4b6d-b83a-8b696de1390b-3ab2de09-1581-4385-91ea-74299b921b9a-v1.png?transparent=1&palette=1&width=246)